packages/core/parcel-bundler/src/assets/ElmAsset.js

```diff
@@ -73,7 +72,9 @@ class ElmAsset extends Asset {
   }
 
   async generate() {
-    let output = this.contents;
+    let compiled = null;
+    compiled = await this.elm.compileToString(this.name, this.elmOpts);
+    let output = compiled.toString();
```

Comment:
To resolve #2147 Elm's compilation step also needed to be moved into this function so that errors are thrown _after_ `collectDependencies` is called.